src/fsipy/automatedPreprocessing/preprocessing_common.py

```diff
@@ -160,3 +143,47 @@ def generate_mesh(surface, add_boundary_layer, meshing_method, solid_thickness,
     mesh = meshGenerator.Mesh
 
     return mesh, remeshSurface
+
+def convert_xml_mesh_to_hdf5(file_name_xml_mesh, scaling_factor=0.001):
+    """Converts an XML mesh to an HDF5 mesh.
+
+    Args:
+        file_name_xml_mesh (str): The name of the XML mesh file.
+        scaling_factor (float, optional): A scaling factor to apply to the mesh coordinates. The default value is 0.001, which converts from millimeters to meters.
+
+    Returns:
+        None
+
+    Raises:
+        FileNotFoundError: If the XML mesh file does not exist.
+    """
+
+    # Check if the XML mesh file exists
+    if not Path(file_name_xml_mesh).is_file():
+        raise FileNotFoundError(f"The file '{file_name_xml_mesh}' does not exist.")
+
+    mesh = Mesh(file_name_xml_mesh)
+
+    # Rescale the mesh coordinates
+    x = mesh.coordinates()
+    x[:, :] *= scaling_factor
+    mesh.bounding_box_tree().build(mesh)
+
+    # Convert subdomains to mesh function
+    boundaries = MeshFunction("size_t", mesh, 2, mesh.domains())
+    boundaries.set_values(boundaries.array() + 1)  # FIXME: Explain why this is necessary
+
+    boundary_file = File(str(Path(file_name_xml_mesh).with_suffix('').with_suffix('')) + '_boundaries.pvd')
```

Comment:
This is to strip away the `.xml.gz` and add `_boundaries` at the end?
I guess this could be done with:
```python
base, first_dot, rest = file_name_xml_mesh.name.partition('.')
filename = file_name_xml_mesh.with_name(base+"_boundaries.pvd")
```